pyingest/tests/test_csgparsers.py

```diff
@@ -0,0 +1,81 @@
+"""
+Test parsers
+"""
+
+import unittest
+import filecmp
+import sys
+import os
+import glob
+import json
+
+from pyingest.parsers import oup
+from pyingest.parsers import iop
+from pyingest.config import config
+from pyingest.parsers.author_names import AuthorNames
+
+from pyingest.serializers import classic
+
+
+
+class TestOUP(unittest.TestCase):
+
+    def setUp(self):
+        stubdata_dir = os.path.join(os.path.dirname(__file__), 'data/stubdata/')
+        self.inputdocs = glob.glob(os.path.join(stubdata_dir, 'input/oup*'))
+        self.outputdir = os.path.join(stubdata_dir, 'parsed')
+        sys.stderr.write("test cases are: {}\n".format(self.inputdocs))
+
+    def test_oup_parser(self):
+        parser = oup.OUPJATSParser()
+        config.REFERENCE_TOPDIR = '/dev/null/'
+        for file in self.inputdocs:
+            # this will raise exceptions if something is wrong
+            with open(file, 'r') as fp:
+                test_data = parser.parse(fp)
+                self.assertIsNotNone(test_data, "%s: error reading doc" % file)
+            basefile = os.path.basename(file)
+            target = os.path.join(self.outputdir, basefile + '.json')
+            # save temporary copy of data structure
+            target_saved = target + '.json'
+            with open(target_saved, 'w') as fp:
+                json.dump(document, fp, sort_keys=True, indent=4)
+
+            ok = False
+            if os.path.exists(target):
+                with open(target, 'r') as fp:
+                    shouldbe = json.load(fp)
+                    self.assertDictEqual(shouldbe, document, "results differ from %s" % target)
+                    ok = True
+            else:
+                sys.stderr.write("could not find shouldbe file %s\n" % target)
+
+            if ok:
+                os.remove(target_saved)
+            else:
+                sys.stderr.write("parsed output dumped in %s\n" % target_saved)
+
+        return
+
+
+class TestIOP(unittest.TestCase):
```

Comment:
We have an IOP parser already in test_parsers.py I think?  (I hope I didn't forget!)